src/domain/app/getLiikuntaStaticProps.ts

```diff
@@ -0,0 +1,110 @@
+import { GetStaticPropsContext, GetStaticPropsResult } from "next";
+import {
+  isApolloError,
+  gql,
+  ApolloClient,
+  NormalizedCacheObject,
+} from "@apollo/client";
+
+import AppConfig from "../../domain/app/AppConfig";
+import { Language } from "../../types";
+import { PAGE_FRAGMENT } from "../../common/components/page/Page";
+import { getMenuLocationFromLanguage } from "../../common/apollo/utils";
+import { staticGenerationLogger } from "../logger";
+import { createCmsApolloClient } from "../clients/cmsApolloClient";
+
+const GLOBAL_QUERY = gql`
+  ${PAGE_FRAGMENT}
+  query PageQuery($menuLocation: MenuLocationEnum!) {
+    ...PageFragment
+    __typename
+  }
+`;
+
+type LiikuntaContext = {
+  cmsClient: ApolloClient<NormalizedCacheObject>;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export default async function getLiikuntaStaticProps<P = Record<string, any>>(
+  context: GetStaticPropsContext,
+  tryToGetPageProps: (
+    liikuntaContext: LiikuntaContext
+  ) => Promise<GetStaticPropsResult<P>>
+) {
+  const cmsClient = createCmsApolloClient();
+
+  try {
+    await getGlobalCMSData({ client: cmsClient, context });
+    const result = await tryToGetPageProps({ cmsClient });
+    const props =
+      "props" in result
+        ? {
+            initialApolloState: cmsClient.cache.extract(),
+            ...result.props,
+          }
+        : undefined;
```

Comment:
Does this mean that if props are not returned from callback, initialApolloState is not extracted?

Reply:
The return value of a `getStaticProps` function looks like this:

```
export type GetStaticPropsResult<P> =
  | { props: P; revalidate?: number | boolean }
  | { redirect: Redirect; revalidate?: number | boolean }
  | { notFound: true; revalidate?: number | boolean }
```

Which leads me to believe, that for a type correct result, we can't inject any fields into `props` if the field does not exist in the return type, because in that case its a return value is of some type that does not have a props field.

As far as I can tell, if the props field is missing, the result has to be a redirect or a not found variant. I assumed that the props will then be extracted when the user lands on a page that does provide props.